eureka-core/src/main/java/com/netflix/eureka/EurekaBootStrap.java

```diff
@@ -82,6 +81,12 @@
 
     protected volatile EurekaServerContext serverContext;
     protected volatile AwsBinder awsBinder;
+    
+    private DiscoveryClient discoveryClient;
```

Comment:
I think it is reasonable to add getEurekaClientConfig() to the EurekaClient interface, which will then allow the reference here to be to the EurekaClient interface instead of the DiscoveryClient implementation.